docs/getting-started.md

```diff
@@ -33,23 +21,288 @@ If you had installed esy previously:
 npm uninstall --global --update esy
 ```
 
-## Clone & initialize the project
+## Step-by-step tutorial
+
+Let's start with an empty `package.json`
+
+```json
+{}
+```
+
+To write a basic hello world program in OCaml, let's add the `ocaml`
+compiler package to the `package.json`. 
+
+```json
+{
+  "dependencies": {
+    "ocaml": "4.10.x"
+  }
+}
+```
+
+And our OCaml source file, `hello.ml` with 
+
+```ml
+let () = print_endline "Hello, World"
+```
+
+To install the dependencies, run `esy`
+
+```sh
+$ esy
+```
+
+This will fetch `ocaml` and install it in a sandbox exclusively
+created for this newly created project (described by `package.json`)
+
+<screenshot>
+
+Just like in the Yarn/NPM world, build commands are run with a prefix.
+
+To compile,
+
+```sh
+esy ocamlopt -o hello.exe hello.ml
+```
+
+To run the binary, `hello.exe`,
+
+```sh
+./hello.exe
+```
+
+### Using a build system
+Taking a step further, to use a build system, [Dune](https://dune.readthedocs.io/en/stable/), and build the same hello
+world program, add `@opam/dune`.
+
+```diff
+  "dependencies": {
+    "ocaml": "4.10.x",
++	"@opam/dune": "*"
+  }
+```
```

Comment:
### Suggested position

6

### Suggested change

Compiling your code with `ocamlopt` or `ocamlc` works well, when you don't have a lot of dependencies. But as soon as your project gets larger, compiling your code manually becomes very complex, very fast. To help with this, we want to use a build system. 

If you are coming from JavaScript, you probably already heard from tools like "Webpack", "Browserify" or "Rollup". These are build systems. They take your code and its dependencies and bundle them together into a single file.

In OCaml, the most popular build system is called [Dune](https://dune.readthedocs.io/en/stable/) and it pretty much does the same thing. You give it an entry point and it _"bundles"_ all of your code and its dependencies together into a single executable.

To install dune into our "Hello World" program, we have to add `@opam/dune` to the `dependencies` of our `package.json`.

_package.json_
```diff
"dependencies": {
    "ocaml": "4.10.x",
+   "@opam/dune": "*"
}
```

We can either do this manually by modifying the file and running `esy install` **or** we can add it with the command `esy add @opam/dune`. The latter automatically installs dune and adds it to our `package.json`.

### Comments / Questions from Review

Again, is it correct to add this to the `dependencies` or would it be better inside of the `devDependencies`?